src/repeated_game.jl

```diff
@@ -525,3 +527,172 @@ function outerapproximation(
 
     return vertices
 end
+
+"""
+    AS(rpd; maxiter=1000, plib=default_library(2, T), tol=1e-5, u=nothing)
+
+Using AS algorithm to compute the set of payoff pairs of all pure-strategy
+subgame-perfect equilibria with public randomization for any repeated
+two-player games with perfect monitoring and discounting, following
+Abreu and Sannikov (2014).
+
+# Arguments
+
+- `rpd::RepeatedGame{2, T}` : Two player repeated game with T<:Real.
+- `maxiter::Integer` : Maximum number of iterations.
+- `plib`: Allows users to choose a particular package for the geometry
+  computations.
+  (See [Polyhedra.jl](https://github.com/JuliaPolyhedra/Polyhedra.jl)
+  docs for more info). By default, it chooses to use SimplePolyhedraLibrary.
+- `tol::Float64` : Tolerance in differences of set.
+- `u` : The punishment payoff pair if any player deviates. In default,
+  we use minimax payoff pair. If there is better guess, you can specify it
+  by passing a `Vector{T}` with length 2.
+
+# Returns
+
+- `::Matrix{T}` : Vertices of the set of payoff pairs.
+"""
+function AS(rpd::RepeatedGame{2, T}; maxiter::Integer=1000,
+            plib=default_library(2, T), tol::Float64=1e-5, u=nothing) where T<:Real
+
+    # Initialize W0 with each entries of payoff bimatrix
+    v_old = _payoff_points(rpd.sg)
+
+    if isnothing(u)
+        u = [minimum(rpd.sg.players[1].payoff_array),
+             minimum(rpd.sg.players[2].payoff_array)]
+    end
+
+    # create VRepresentation and Polyhedron and get rid of redundant vertices
+    p = polyhedron(vrep(v_old), plib)
+    removevredundancy!(p)
+    H = hrep(p)
+
+    # calculate the best deviation gains
+    # normalize with (1-delta)/delta
+    best_dev_gains1, best_dev_gains2 = (1-rpd.delta)/rpd.delta .* _best_dev_gains(rpd.sg)
+
+    for iter = 1:maxiter
+
+        v_new = T[] # to store new vertices
```

Comment:
If `T == Int`, then this will be a vector of `Int`s, which will lead to `InexactError`.